src/clientSideScene/sceneEntities.ts

```diff
@@ -1584,6 +1667,448 @@ export class SceneEntities {
       expressionIndexToDelete: insertIndex,
     }
   }
+  setupDraftArc = async (
+    sketchEntryNodePath: PathToNode,
+    sketchNodePaths: PathToNode[],
+    planeNodePath: PathToNode,
+    forward: [number, number, number],
+    up: [number, number, number],
+    sketchOrigin: [number, number, number],
+    center: [x: number, y: number]
+  ): Promise<SketchDetailsUpdate | Error> => {
+    let _ast = structuredClone(kclManager.ast)
+
+    const _node1 = getNodeFromPath<VariableDeclaration>(
+      _ast,
+      sketchEntryNodePath || [],
+      'VariableDeclaration'
+    )
+    if (trap(_node1)) return Promise.reject(_node1)
+    const variableDeclarationName = _node1.node?.declaration.id?.name || ''
+
+    const sg = sketchFromKclValue(
+      kclManager.variables[variableDeclarationName],
+      variableDeclarationName
+    )
+    if (err(sg)) return Promise.reject(sg)
+    const lastSeg = sg?.paths?.slice(-1)[0] || sg.start
+
+    // Calculate a default center point and radius based on the last segment's endpoint
+    const from: [number, number] = [lastSeg.to[0], lastSeg.to[1]]
+    const radius = Math.sqrt(
+      Math.pow(center[0] - from[0], 2) + Math.pow(center[1] - from[1], 2)
```

Comment:
Why not `** 2` as done elsewhere in this PR?

Reply:
3ad5118f5020f86421fc87f9868f7544b8e6e158